TICC.ino

```diff
@@ -16,17 +16,24 @@ int int_pin = 3; // physical pin
 volatile unsigned long long coarse_count = 0;
 
 // hardware connections to TDC2700 -- need to reassign these more logically (todo in Eagle)
-const int ENABLE_A = 13;
-const int INTB_A = 12;
-const int CSB_A = 11;
-const int ENABLE_B = 10;
-const int INTB_B = 9;
-const int CSB_B = 8;
-const int STOP_A = 7;
-const int STOP_B = 8;
-
-const int ChA = 0;	// channel A
-const int ChB = 1;	// channel B
+
+struct chanTDC7200_t {
+  const int ENABLE;
+  const int INTB;
+  const int CSB;
+  const int STOP;
+  int timeresult;
+  int clockresult;
+  int calresult;
+  long result;
+  long long unsigned stop_time;
+};
+
+static chanTDC7200_t ChanA = { 13, 12, 11, 6, 0, 0, 0, 0, 0 }; // I/O pins on Arduino
+static chanTDC7200_t ChanB = { 10, 9, 8, 7, 0, 0, 0, 0, 0 };
```

Comment:
This I like!  I'm too used to old-style C to have been aware that named initializers snuck in, or snuck in to Arduino.
George

-----Original Message-----
From: Jeremy McDermond [mailto:notifications@github.com] 
Sent: Monday, April 25, 2016 10:29 PM
To: TAPR/TICC <TICC@noreply.github.com>
Subject: Re: [TAPR/TICC] Refactor channel to a structure, make it easier to add more channels. (#2)

In TICC.ino <https://github.com/TAPR/TICC/pull/2#discussion_r61021006> :

> -const int ChB = 1;	// channel B
> +
> +struct chanTDC7200_t {
> +  const int ENABLE;
> +  const int INTB;
> +  const int CSB;
> +  const int STOP;
> +  int timeresult;
> +  int clockresult;
> +  int calresult;
> +  long result;
> +  long long unsigned stop_time;
> +};
> +
> +static chanTDC7200_t ChanA = { 13, 12, 11, 6, 0, 0, 0, 0, 0 }; // I/O pins on Arduino
> +static chanTDC7200_t ChanB = { 10, 9, 8, 7, 0, 0, 0, 0, 0 };

When you use initializers like this, it's more clear if you do something like:

static chanTDC7200_t ChanA = {
    .ENABLE = 13,
    .INTB = 12,
    .CSB = 11,
    .STOP = 6,
    .timeresult = 0,
    .clockresult = 0,
    .calresult = 0,
    .result = 0,
    .stop_time = 0,
};

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub <https://github.com/TAPR/TICC/pull/2/files/7ee79603f41f5977d8b78f810d2bcc69cab948e7#r61021006>  <https://github.com/notifications/beacon/AKfFAL4qiL3MB71PHmMlfs0vi4oMDa5vks5p7XhbgaJpZM4IPh4v.gif>